compiler/ast.nim

```diff
@@ -1581,7 +1581,7 @@ proc hasPattern*(s: PSym): bool {.inline.} =
   result = isRoutine(s) and s.ast.sons[patternPos].kind != nkEmpty
 
 iterator items*(n: PNode): PNode =
-  for i in 0.. <n.len: yield n.sons[i]
+  for i in 0.. <n.safeLen: yield n.sons[i]
```

Comment:
that should be a ``safeItems`` iterator then? I dislike code like this, the client should know enough about the node kind to at least be able to say whether it's an atomic node or not.